### ISSUE TITLE
branch has 0 operands for successor #0, but target block has 1

### ISSUE THREAD

**Issue by KronicDeth** [lumen/lumen#534]
## Reproduction steps

1. `git checkout a7067fa548f38e745dd861a03ed4de0ebee74b21`
2. `cargo test --package liblumen_otp erlang::binary_to_list_3::with_binary_with_start_less_than_or_equal_to_stop_returns_list_of_bytes`

## Error

```
 Compiling tests/lib/erlang/binary_to_list_3/with_binary_with_start_less_than_or_equal_to_stop_returns_list_of_bytes/init.erl
loc("tests/lib/erlang/binary_to_list_3/with_binary_with_start_less_than_or_equal_to_stop_returns_list_of_bytes/init.erl":12:36): error: branch has 0 operands for successor #0, but target block has 1


module @init {
  llvm.func @"erlang:display/1"(!llvm.i64) -> !llvm.i64
  llvm.func @"erlang:binary_to_list/3"(!llvm.i64, !llvm.i64, !llvm.i64) -> !llvm.i64
  llvm.func @__lumen_builtin_encode_immediate(!llvm.i32, !llvm.i64) -> !llvm.i64
  llvm.func @__lumen_builtin_math.add(!llvm.i64, !llvm.i64) -> !llvm.i64
  llvm.func @__lumen_builtin_cmp.eq(!llvm.i64, !llvm.i64) -> !llvm.i1
  llvm.func @__lumen_get_exception(!llvm<"i8*">) -> !llvm.i64 attributes {argmemonly, nounwind, readonly}
  llvm.func @"erlang:byte_size/1"(!llvm.i64) -> !llvm.i64
  llvm.func @__lumen_builtin_binary_finish(!llvm.i64) -> !llvm.i64
  llvm.func @__lumen_builtin_binary_push_integer(!llvm.i64, !llvm.i64, !llvm.i64, !llvm.i8, !llvm.i1, !llvm.i32) -> !llvm<"%binary.pushed = type { i64, i1 }">
  llvm.func @__lumen_builtin_binary_start() -> !llvm.i64
  llvm.func @__lumen_builtin_yield()
  llvm.mlir.global external local_exec @CURRENT_REDUCTION_COUNT() : !llvm.i32
  llvm.func @lumen_eh_personality(...) -> !llvm.i32
  llvm.func @"init:start/0"() -> !llvm.i64 attributes {personality = @lumen_eh_personality} {
    %0 = llvm.mlir.constant(20 : i32) : !llvm.i32
    %1 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %2 = llvm.load %1 : !llvm<"i32*">
    %3 = llvm.icmp "uge" %2, %0 : !llvm.i32
    llvm.cond_br %3, ^bb1, ^bb2
  ^bb1:  // pred: ^bb0
    llvm.call @__lumen_builtin_yield() : () -> ()
    llvm.br ^bb2
  ^bb2:  // 2 preds: ^bb0, ^bb1
    %4 = llvm.call @__lumen_builtin_binary_start() : () -> !llvm.i64
    %5 = llvm.mlir.constant(140737488355328 : i64) : !llvm.i64
    %6 = llvm.mlir.constant(140737488355336 : i64) : !llvm.i64
    %7 = llvm.mlir.constant(1 : i8) : !llvm.i8
    %8 = llvm.mlir.constant(false) : !llvm.i1
    %9 = llvm.mlir.constant(0 : i32) : !llvm.i32
    %10 = llvm.call @__lumen_builtin_binary_push_integer(%4, %5, %6, %7, %8, %9) : (!llvm.i64, !llvm.i64, !llvm.i64, !llvm.i8, !llvm.i1, !llvm.i32) -> !llvm<"%binary.pushed = type { i64, i1 }">
    %11 = llvm.extractvalue %10[0] : !llvm<"%binary.pushed = type { i64, i1 }">
    %12 = llvm.extractvalue %10[1] : !llvm<"%binary.pushed = type { i64, i1 }">
    %13 = llvm.mlir.constant(140737488355329 : i64) : !llvm.i64
    %14 = llvm.mlir.constant(140737488355336 : i64) : !llvm.i64
    %15 = llvm.mlir.constant(1 : i8) : !llvm.i8
    %16 = llvm.mlir.constant(false) : !llvm.i1
    %17 = llvm.mlir.constant(0 : i32) : !llvm.i32
    %18 = llvm.call @__lumen_builtin_binary_push_integer(%11, %13, %14, %15, %16, %17) : (!llvm.i64, !llvm.i64, !llvm.i64, !llvm.i8, !llvm.i1, !llvm.i32) -> !llvm<"%binary.pushed = type { i64, i1 }">
    %19 = llvm.extractvalue %18[0] : !llvm<"%binary.pushed = type { i64, i1 }">
    %20 = llvm.extractvalue %18[1] : !llvm<"%binary.pushed = type { i64, i1 }">
    %21 = llvm.mlir.constant(140737488355330 : i64) : !llvm.i64
    %22 = llvm.mlir.constant(140737488355336 : i64) : !llvm.i64
    %23 = llvm.mlir.constant(1 : i8) : !llvm.i8
    %24 = llvm.mlir.constant(false) : !llvm.i1
    %25 = llvm.mlir.constant(0 : i32) : !llvm.i32
    %26 = llvm.call @__lumen_builtin_binary_push_integer(%19, %21, %22, %23, %24, %25) : (!llvm.i64, !llvm.i64, !llvm.i64, !llvm.i8, !llvm.i1, !llvm.i32) -> !llvm<"%binary.pushed = type { i64, i1 }">
    %27 = llvm.extractvalue %26[0] : !llvm<"%binary.pushed = type { i64, i1 }">
    %28 = llvm.extractvalue %26[1] : !llvm<"%binary.pushed = type { i64, i1 }">
    %29 = llvm.call @__lumen_builtin_binary_finish(%27) : (!llvm.i64) -> !llvm.i64
    %30 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %31 = llvm.mlir.constant(1 : i32) : !llvm.i32
    %32 = llvm.atomicrmw add %30, %31 monotonic  : !llvm.i32
    %33 = llvm.call @"init:lists/1"(%29) {musttail} : (!llvm.i64) -> !llvm.i64
    llvm.return %33 : !llvm.i64
  }
  llvm.func @"init:lists/1"(%arg0: !llvm.i64) -> !llvm.i64 attributes {personality = @lumen_eh_personality} {
    %0 = llvm.mlir.constant(20 : i32) : !llvm.i32
    %1 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %2 = llvm.load %1 : !llvm<"i32*">
    %3 = llvm.icmp "uge" %2, %0 : !llvm.i32
    llvm.cond_br %3, ^bb1, ^bb2
  ^bb1:  // pred: ^bb0
    llvm.call @__lumen_builtin_yield() : () -> ()
    llvm.br ^bb2
  ^bb2:  // 2 preds: ^bb0, ^bb1
    %4 = llvm.mlir.constant(140737488355329 : i64) : !llvm.i64
    %5 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %6 = llvm.mlir.constant(1 : i32) : !llvm.i32
    %7 = llvm.atomicrmw add %5, %6 monotonic  : !llvm.i32
    %8 = llvm.call @"init:lists/2"(%arg0, %4) {musttail} : (!llvm.i64, !llvm.i64) -> !llvm.i64
    llvm.return %8 : !llvm.i64
  }
  llvm.func @"init:lists/2"(%arg0: !llvm.i64, %arg1: !llvm.i64) -> !llvm.i64 attributes {personality = @lumen_eh_personality} {
    %0 = llvm.mlir.constant(20 : i32) : !llvm.i32
    %1 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %2 = llvm.load %1 : !llvm<"i32*">
    %3 = llvm.icmp "uge" %2, %0 : !llvm.i32
    llvm.cond_br %3, ^bb1, ^bb2
  ^bb1:  // pred: ^bb0
    llvm.call @__lumen_builtin_yield() : () -> ()
    llvm.br ^bb2
  ^bb2:  // 2 preds: ^bb0, ^bb1
    %4 = llvm.mlir.null : !llvm<"i8*">
    %5 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %6 = llvm.mlir.constant(1 : i32) : !llvm.i32
    %7 = llvm.atomicrmw add %5, %6 monotonic  : !llvm.i32
    llvm.invoke @"erlang:byte_size/1"(%arg0) to ^bb3 unwind ^bb7 : (!llvm.i64) -> ()
  ^bb3(%8: !llvm.i64):  // pred: ^bb2
    %9 = llvm.call @__lumen_builtin_cmp.eq(%arg1, %8) : (!llvm.i64, !llvm.i64) -> !llvm.i1
    %10 = llvm.zext %9 : !llvm.i1 to !llvm.i64
    %11 = llvm.mlir.constant(6 : i32) : !llvm.i32
    %12 = llvm.call @__lumen_builtin_encode_immediate(%11, %10) : (!llvm.i32, !llvm.i64) -> !llvm.i64
    llvm.br ^bb5(%12 : !llvm.i64)
  ^bb4:  // pred: ^bb5
    %13 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %14 = llvm.mlir.constant(1 : i32) : !llvm.i32
    %15 = llvm.atomicrmw add %13, %14 monotonic  : !llvm.i32
    %16 = llvm.call @"init:lists/3"(%arg0, %arg1, %arg1) {musttail} : (!llvm.i64, !llvm.i64, !llvm.i64) -> !llvm.i64
    llvm.return %16 : !llvm.i64
  ^bb5(%17: !llvm.i64):  // 2 preds: ^bb3, ^bb8
    %18 = llvm.mlir.constant(562949953421313 : i64) : !llvm.i64
    %19 = llvm.call @__lumen_builtin_cmp.eq(%17, %18) : (!llvm.i64, !llvm.i64) -> !llvm.i1
    llvm.cond_br %19, ^bb4, ^bb6
  ^bb6:  // pred: ^bb5
    %20 = llvm.mlir.constant(562949953421312 : i64) : !llvm.i64
    %21 = llvm.call @__lumen_builtin_cmp.eq(%17, %20) : (!llvm.i64, !llvm.i64) -> !llvm.i1
    %22 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %23 = llvm.mlir.constant(1 : i32) : !llvm.i32
    %24 = llvm.atomicrmw add %22, %23 monotonic  : !llvm.i32
    %25 = llvm.call @"init:lists/3"(%arg0, %arg1, %arg1) {tail} : (!llvm.i64, !llvm.i64, !llvm.i64) -> !llvm.i64
    %26 = llvm.mlir.constant(140737488355329 : i64) : !llvm.i64
    %27 = llvm.call @__lumen_builtin_math.add(%arg1, %26) : (!llvm.i64, !llvm.i64) -> !llvm.i64
    %28 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %29 = llvm.mlir.constant(1 : i32) : !llvm.i32
    %30 = llvm.atomicrmw add %28, %29 monotonic  : !llvm.i32
    %31 = llvm.call @"init:lists/2"(%arg0, %27) {musttail} : (!llvm.i64, !llvm.i64) -> !llvm.i64
    llvm.return %31 : !llvm.i64
  ^bb7:  // pred: ^bb2
    %32 = llvm.landingpad (catch %4 : !llvm<"i8*">) : !llvm<"%lumen.exception = type { i8*, i32 }">
    %33 = llvm.extractvalue %32[0] : !llvm<"%lumen.exception = type { i8*, i32 }">
    llvm.br ^bb8
  ^bb8:  // pred: ^bb7
    %34 = llvm.call @__lumen_get_exception(%33) {tail} : (!llvm<"i8*">) -> !llvm.i64
    %35 = llvm.inttoptr %34 : !llvm.i64 to !llvm<"%"type 0x7ff1ae820140"*">
    %36 = llvm.mlir.constant(0 : i32) : !llvm.i32
    %37 = llvm.mlir.constant(1 : i32) : !llvm.i32
    %38 = llvm.getelementptr %35[%36, %37] : (!llvm<"%"type 0x7ff1ae820140"*">, !llvm.i32, !llvm.i32) -> !llvm<"i64*">
    %39 = llvm.load %38 : !llvm<"i64*">
    %40 = llvm.mlir.constant(2 : i32) : !llvm.i32
    %41 = llvm.getelementptr %35[%36, %40] : (!llvm<"%"type 0x7ff1ae820140"*">, !llvm.i32, !llvm.i32) -> !llvm<"i64*">
    %42 = llvm.load %41 : !llvm<"i64*">
    %43 = llvm.mlir.constant(3 : i32) : !llvm.i32
    %44 = llvm.getelementptr %35[%36, %43] : (!llvm<"%"type 0x7ff1ae820140"*">, !llvm.i32, !llvm.i32) -> !llvm<"i64*">
    %45 = llvm.load %44 : !llvm<"i64*">
    %46 = llvm.mlir.constant(562949953421312 : i64) : !llvm.i64
    llvm.br ^bb5(%46 : !llvm.i64)
  }
  llvm.func @"init:lists/3"(%arg0: !llvm.i64, %arg1: !llvm.i64, %arg2: !llvm.i64) -> !llvm.i64 attributes {personality = @lumen_eh_personality} {
    %0 = llvm.mlir.constant(20 : i32) : !llvm.i32
    %1 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %2 = llvm.load %1 : !llvm<"i32*">
    %3 = llvm.icmp "uge" %2, %0 : !llvm.i32
    llvm.cond_br %3, ^bb1, ^bb2
  ^bb1:  // pred: ^bb0
    llvm.call @__lumen_builtin_yield() : () -> ()
    llvm.br ^bb2
  ^bb2:  // 2 preds: ^bb0, ^bb1
    %4 = llvm.mlir.null : !llvm<"i8*">
    %5 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %6 = llvm.mlir.constant(1 : i32) : !llvm.i32
    %7 = llvm.atomicrmw add %5, %6 monotonic  : !llvm.i32
    llvm.invoke @"erlang:byte_size/1"(%arg0) to ^bb3 unwind ^bb7 : (!llvm.i64) -> ()
  ^bb3(%8: !llvm.i64):  // pred: ^bb2
    %9 = llvm.call @__lumen_builtin_cmp.eq(%arg2, %8) : (!llvm.i64, !llvm.i64) -> !llvm.i1
    %10 = llvm.zext %9 : !llvm.i1 to !llvm.i64
    %11 = llvm.mlir.constant(6 : i32) : !llvm.i32
    %12 = llvm.call @__lumen_builtin_encode_immediate(%11, %10) : (!llvm.i32, !llvm.i64) -> !llvm.i64
    llvm.br ^bb5(%12 : !llvm.i64)
  ^bb4:  // pred: ^bb5
    %13 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %14 = llvm.mlir.constant(1 : i32) : !llvm.i32
    %15 = llvm.atomicrmw add %13, %14 monotonic  : !llvm.i32
    %16 = llvm.call @"init:list/3"(%arg0, %arg1, %arg2) {musttail} : (!llvm.i64, !llvm.i64, !llvm.i64) -> !llvm.i64
    llvm.return %16 : !llvm.i64
  ^bb5(%17: !llvm.i64):  // 2 preds: ^bb3, ^bb8
    %18 = llvm.mlir.constant(562949953421313 : i64) : !llvm.i64
    %19 = llvm.call @__lumen_builtin_cmp.eq(%17, %18) : (!llvm.i64, !llvm.i64) -> !llvm.i1
    llvm.cond_br %19, ^bb4, ^bb6
  ^bb6:  // pred: ^bb5
    %20 = llvm.mlir.constant(562949953421312 : i64) : !llvm.i64
    %21 = llvm.call @__lumen_builtin_cmp.eq(%17, %20) : (!llvm.i64, !llvm.i64) -> !llvm.i1
    %22 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %23 = llvm.mlir.constant(1 : i32) : !llvm.i32
    %24 = llvm.atomicrmw add %22, %23 monotonic  : !llvm.i32
    %25 = llvm.call @"init:list/3"(%arg0, %arg1, %arg2) {tail} : (!llvm.i64, !llvm.i64, !llvm.i64) -> !llvm.i64
    %26 = llvm.mlir.constant(140737488355329 : i64) : !llvm.i64
    %27 = llvm.call @__lumen_builtin_math.add(%arg2, %26) : (!llvm.i64, !llvm.i64) -> !llvm.i64
    %28 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %29 = llvm.mlir.constant(1 : i32) : !llvm.i32
    %30 = llvm.atomicrmw add %28, %29 monotonic  : !llvm.i32
    %31 = llvm.call @"init:lists/3"(%arg0, %arg1, %27) {musttail} : (!llvm.i64, !llvm.i64, !llvm.i64) -> !llvm.i64
    llvm.return %31 : !llvm.i64
  ^bb7:  // pred: ^bb2
    %32 = llvm.landingpad (catch %4 : !llvm<"i8*">) : !llvm<"%lumen.exception = type { i8*, i32 }">
    %33 = llvm.extractvalue %32[0] : !llvm<"%lumen.exception = type { i8*, i32 }">
    llvm.br ^bb8
  ^bb8:  // pred: ^bb7
    %34 = llvm.call @__lumen_get_exception(%33) {tail} : (!llvm<"i8*">) -> !llvm.i64
    %35 = llvm.inttoptr %34 : !llvm.i64 to !llvm<"%"type 0x7ff1ae820270"*">
    %36 = llvm.mlir.constant(0 : i32) : !llvm.i32
    %37 = llvm.mlir.constant(1 : i32) : !llvm.i32
    %38 = llvm.getelementptr %35[%36, %37] : (!llvm<"%"type 0x7ff1ae820270"*">, !llvm.i32, !llvm.i32) -> !llvm<"i64*">
    %39 = llvm.load %38 : !llvm<"i64*">
    %40 = llvm.mlir.constant(2 : i32) : !llvm.i32
    %41 = llvm.getelementptr %35[%36, %40] : (!llvm<"%"type 0x7ff1ae820270"*">, !llvm.i32, !llvm.i32) -> !llvm<"i64*">
    %42 = llvm.load %41 : !llvm<"i64*">
    %43 = llvm.mlir.constant(3 : i32) : !llvm.i32
    %44 = llvm.getelementptr %35[%36, %43] : (!llvm<"%"type 0x7ff1ae820270"*">, !llvm.i32, !llvm.i32) -> !llvm<"i64*">
    %45 = llvm.load %44 : !llvm<"i64*">
    %46 = llvm.mlir.constant(562949953421312 : i64) : !llvm.i64
    llvm.br ^bb5(%46 : !llvm.i64)
  }
  llvm.func @"init:list/3"(%arg0: !llvm.i64, %arg1: !llvm.i64, %arg2: !llvm.i64) -> !llvm.i64 attributes {personality = @lumen_eh_personality} {
    %0 = llvm.mlir.constant(20 : i32) : !llvm.i32
    %1 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %2 = llvm.load %1 : !llvm<"i32*">
    %3 = llvm.icmp "uge" %2, %0 : !llvm.i32
    llvm.cond_br %3, ^bb1, ^bb2
  ^bb1:  // pred: ^bb0
    llvm.call @__lumen_builtin_yield() : () -> ()
    llvm.br ^bb2
  ^bb2:  // 2 preds: ^bb0, ^bb1
    %4 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %5 = llvm.mlir.constant(1 : i32) : !llvm.i32
    %6 = llvm.atomicrmw add %4, %5 monotonic  : !llvm.i32
    %7 = llvm.call @"erlang:binary_to_list/3"(%arg0, %arg1, %arg2) {tail} : (!llvm.i64, !llvm.i64, !llvm.i64) -> !llvm.i64
    %8 = llvm.mlir.addressof @CURRENT_REDUCTION_COUNT : !llvm<"i32*">
    %9 = llvm.mlir.constant(1 : i32) : !llvm.i32
    %10 = llvm.atomicrmw add %8, %9 monotonic  : !llvm.i32
    %11 = llvm.call @"erlang:display/1"(%7) {musttail} : (!llvm.i64) -> !llvm.i64
    llvm.return %11 : !llvm.i64
  }
}error: lowering to mlir (llvm dialect) failed
```

The `loc` referenced is

```
lists(Binary, Start) when Start == byte_size(Binary) ->
```

## `init.erl`

```erlang
-module(init).
-export([start/0]).
-import(erlang, [binary_to_list/3, display/1]).

start() ->
  Binary = <<0, 1, 2>>,
  lists(Binary).

lists(Binary) ->
  lists(Binary, 1).

lists(Binary, Start) when Start == byte_size(Binary) ->
  lists(Binary, Start, Start);
lists(Binary, Start) ->
  lists(Binary, Start, Start),
  lists(Binary, Start + 1).

lists(Binary, Start, Stop) when Stop == byte_size(Binary) ->
  list(Binary, Start, Stop);
lists(Binary, Start, Stop) ->
  list(Binary, Start, Stop),
  lists(Binary, Start, Stop + 1).

list(Binary, Start, Stop) ->
  display(binary_to_list(Binary, Start, Stop)).

```

**Comment by KronicDeth** [comment]
Switching to a `case` in the body instead of a `guard` on the function clauses is a work-around

## `init.erl`

```erlang
-module(init).
-export([start/0]).
-import(erlang, [binary_to_list/3, display/1]).

start() ->
  Binary = <<0, 1, 2>>,
  lists(Binary).

lists(Binary) ->
  lists(Binary, 1).

lists(Binary, Start) ->
  case byte_size(Binary) of
    Start ->
      lists(Binary, Start, Start);
    _ ->
      lists(Binary, Start, Start),
      lists(Binary, Start + 1)
  end.

lists(Binary, Start, Stop) ->
  case byte_size(Binary) of
    Stop ->
      list(Binary, Start, Stop);
    _ ->
      list(Binary, Start, Stop),
      lists(Binary, Start, Stop + 1)
  end.

list(Binary, Start, Stop) ->
  display(binary_to_list(Binary, Start, Stop)).
```

**Comment by KronicDeth** [comment]
This also triggers the errors

## `init.erl`

```erlanng
-module(init).
-export([start/0]).
-import(erlang, [display/1]).
-import(lumen, [is_big_integer/1, is_small_integer/1]).

start() ->
  errors(atom()),
  errors(big_integer()),
  errors(binary()),
  errors(empty_list()),
  errors(float()),
  errors(map()),
  errors(non_empty_list()),
  errors(pid()),
  errors(reference()),
  errors(small_integer()),
  errors(tuple()).

errors(Fun) ->
  {'EXIT',{Reason, _Stacktrace}} = (catch spawn(Fun)),
  display(Reason).

atom() ->
  atom.

big_integer() ->
  BigInteger = (1 bsl 63),
  true = is_big_integer(BigInteger),
  BigInteger.

binary() ->
  <<"binary">>.

empty_list() ->
  [].

float() ->
  0.0.

%% work around MapPut not implemented
map() ->
  maps:from_list([]).

non_empty_list() ->
  [non_empty].

pid() ->
  self().

reference() ->
  make_ref().

small_integer() ->
  SmallInteger = 0,
  true = is_small_integer(SmallInteger),
  SmallInteger.

tuple() ->
  [].
```

## Error

```
loc("tests/lib/erlang/spawn_1/without_function_errors_badarg/init.erl":20:43): error: branch has 0 operands for successor #0, but target block has 1
```

That line is `{'EXIT',{Reason, _Stacktrace}} = (catch spawn(Fun)),`

**Comment by KronicDeth** [comment]
It also occurs with a `try catch`:

## `init.erl`

```erlang
-module(init).
-export([start/0]).
-import(erlang, [display/1]).
-import(lumen, [is_big_integer/1, is_small_integer/1]).

start() ->
  errors(atom()),
  errors(big_integer()),
  errors(binary()),
  errors(empty_list()),
  errors(float()),
  errors(map()),
  errors(non_empty_list()),
  errors(pid()),
  errors(reference()),
  errors(small_integer()),
  errors(tuple()).

errors(Fun) ->
  Try = try spawn(Fun) of
    Spawned -> {spawned, Spawned}
  catch
    Exception -> Exception
  end,
  display(Try).

atom() ->
  atom.

big_integer() ->
  BigInteger = (1 bsl 63),
  true = is_big_integer(BigInteger),
  BigInteger.

binary() ->
  <<"binary">>.

empty_list() ->
  [].

float() ->
  0.0.

%% work around MapPut not implemented
map() ->
  maps:from_list([]).

non_empty_list() ->
  [non_empty].

pid() ->
  self().

reference() ->
  make_ref().

small_integer() ->
  SmallInteger = 0,
  true = is_small_integer(SmallInteger),
  SmallInteger.

tuple() ->
  [].
```

## Error

```
loc("tests/lib/erlang/spawn_1/without_function_errors_badarg/init.erl":20:13): error: branch has 0 operands for successor #0, but target block has 1
```

which is

```
Try = try spawn(Fun) of
```

**Comment by bitwalker** [comment]
Fixed in #582
